lib/rpmsg/rpmsg_virtio.c

```diff
@@ -277,10 +277,10 @@ static int rpmsg_virtio_wait_remote_ready(struct rpmsg_virtio_device *rvdev)
 	uint8_t status;
 
 	while (1) {
-		status = rpmsg_virtio_get_status(rvdev);
+		virtio_get_status(rvdev->vdev, &status);
```

Comment:
Yes we should, but not checking for error was already the case before.

We actually shouldn't need this function at all as waiting for this flag is already done at this point by `rproc_virtio_wait_remote_ready()` which is the same function. I am going to remove this function in a patch I will send after this is taken. At that point I will add checks and propagate the error up the stack. One fix at a time :smile:

Reply:
> Yes we should, but not checking for error was already the case before.

The difference is that before `rproc_virtio_get_status `returns only the status, `virtio_get_status `test the input parameters

> 
> We actually shouldn't need this function at all as waiting for this flag is already done at this point by `rproc_virtio_wait_remote_ready()` which is the same function. I am going to remove this function in a patch I will send after this is taken. At that point I will add checks and propagate the error up the stack. One fix at a time 😄

The RPMsg and remoteproc_virtio  are two layers that can be used independently. remoteproc_virtio is not necessarily used; some projects define their own transport layer and use this function to synchronize the communication. Therefore, this one should be kept.

Reply:
This check belongs to the virtio layer. There are two function that do the same thing currently (`rproc_virtio_wait_remote_ready()` and `rpmsg_virtio_wait_remote_ready()`) what I plan to do is merge them into one that lives in the virtio layer. Then any layer that uses virtio can make use of that.

Reply:
`rpmsg_virtio_wait_remote_ready ` name is probably not good. the name should be something such as `rpmsg_virtio_wait_ready`.  
 
There are two functions because RPMSG layers and RPROC virtio backend are independent and must be kept independant .
here is an exemple of projet that uses rpmsg but not rproc_virtio:
https://elixir.bootlin.com/linux/latest/source/drivers/virtio/virtio_ring.c

If you want to merge, the service should be implemented in the `virtio `layer. But does it makes sense to add a virtio API for that? I would prefer not to.

Reply:
I think we can figure that out later then, for now to keep this PR moving I'll just make the status check fixes as suggested.